src/app-layer-parser.c

```diff
@@ -140,6 +140,8 @@ struct AppLayerParserState_ {
      * we don't need a var per direction since we don't log a transaction
      * unless we have the entire transaction. */
     uint64_t log_id;
+    /* Count of number of transactions */
+    uint64_t tx_cnt;
```

Comment:
I don't follow this. We can take the current tx_cnt at the start of AppLayerParse, and again at the end of it. Then consider the diff to be the new tx number?